lib/server.js

```diff
@@ -0,0 +1,27 @@
+var Router = require(__dirname + '/router.js');
+var http = require('http');
+var url = require('url');
+var port = process.env.PORT || 3000;
+
+var theRouter = new Router();
+
+function httpRequest(req, res) {
```

Comment:
This is sort of the plan I was thinking of.  It would have some sort of call to pass the req function into the route, like ``` router(req, res) ```.  The router module will read the route and method from req, process a callback, and update the res object to be written.